src/Security/UserRoleResolver.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Security;
+
+use Ibexa\Contracts\Core\Repository\Exceptions\InvalidArgumentException;
+use Ibexa\Contracts\Core\Repository\Repository;
+use Ibexa\Contracts\Core\Repository\Values\Content\LocationQuery;
+use Ibexa\Contracts\Core\Repository\Values\Content\Query\Criterion;
+use Ibexa\Contracts\Core\Repository\Values\Content\Search\SearchResult;
+use Ibexa\Contracts\Core\Repository\Values\User\User;
+use Ibexa\Contracts\Core\SiteAccess\ConfigResolverInterface;
+use Netgen\IbexaSiteApi\API\FilterService;
+use Netgen\IbexaSiteApi\API\LoadService;
+use Netgen\IbexaSiteApi\API\Values\Location;
+use Netgen\IbexaSiteApi\Core\Traits\SearchResultExtractorTrait;
+
+use function array_filter;
+use function array_map;
+use function array_unique;
+use function count;
+use function in_array;
+
+final class UserRoleResolver
```

Comment:
Is this really related to the impersonation module?

Reply:
Besides, it is missing the login listener which actually runs the resolver.

Reply:
I added it while we had the login listener in the project and forgot to remove it after I deleted the listener. Deleted now https://github.com/netgen/media-site/pull/194/commits/0c730f88e1cced16467dbb166e16624ac37fccc8